keystone/gauss_fit.py

```diff
@@ -0,0 +1,230 @@
+    	Parameters
+    	----------
+    	region : str
+        	Name of region to reduce
+    	snr_min : float
+        	Lowest signal-to-noise pixels to include in the line-fitting
+    	mol : str
+        	name of molecule to fit
+   	vmin : numpy.float
+        	Minimum centroid velocity, in km/s.
+    	vmax : numpy.float
+        	Maximum centroid velocity, in km/s.
+    	convolve : bool or float
+        	If not False, specifies the beam-size to convolve the original map with
+		Beam-size must be given in arcseconds
+    	use_old_conv : bool
+        	If True, use an already convolved map with name:
+		region + '_' + mol + file_extension + '_conv.fits'
+		This convolved map must be in units of km/s
+    	multicore : int
+		Maximum number of simultaneous processes desired
+	file_extension: str
+		filename extension 
+    	"""
+    	if file_extension:
+        	root = file_extension
+    	else:
+        	# root = 'base{0}'.format(blorder)
+        	root = 'all'
+
+	molecules = ['C2S', 'HC7N_22_21', 'HC7N_21_20', 'HC5N']
+
+    	MolFile = '{0}/{0}_{2}_{1}.fits'.format(region,root,mol)
+	ConvFile = '{0}/{0}_{2}_{1}_conv.fits'.format(region,root,mol)
+	GaussOut = '{0}/{0}_{2}_{1}_gauss_cube.fits'.format(region,root,mol)
+	GaussNoiseOut = '{0}/{0}_{2}_{1}_gauss_cube_noise.fits'.format(region,root,mol)
+	ParamOut = '{0}/{0}_{2}_{1}_param_cube.fits'.format(region,root,mol)
+
+	# Load the spectral cube and convert to velocity units
+	cube = SpectralCube.read(MolFile)
+	cube_km = cube.with_spectral_unit(u.km / u.s, velocity_convention='radio')
+
+	# If desired, convolve map with larger beam 
+	# or load previously created convolved cube
+	if convolve!=False:
+		cube = SpectralCube.read(MolFile)
+		cube_km_1 = cube.with_spectral_unit(u.km / u.s, velocity_convention='radio')
+		beam = radio_beam.Beam(major=convolve*u.arcsec, minor=convolve*u.arcsec, pa=0*u.deg)
+		cube_km = cube_km_1.convolve_to(beam)
+		cube_km.write(ConvFile, format='fits', overwrite=True)
+	if use_old_conv!=False:
+		cube_km = SpectralCube.read(ConvFile)
+	
+	# Define the spectral axis in km/s
+	spectra_x_axis_kms = np.array(cube_km.spectral_axis) 
+
+	# Find the channel range corresponding to vmin and vmax
+	# -- This is a hold-over from when I originally set up the code to 
+	#    use a channel range rather than velocity range.
+	#    Can change later, but this should work for now. 
+	low_channel = np.where(spectra_x_axis_kms<=vmax)[0][0]+1 # Add ones to change index to channel
+	high_channel = np.where(spectra_x_axis_kms>=vmin)[0][-1]+1 # Again, hold-over from older setup
+	peak_channels = [low_channel, high_channel]
+
+	# Create cubes for storing the fitted Gaussian profiles
+	# and the Gaussians with noise added back into the spectrum
+	header = cube_km.header
+	cube_gauss = np.array(cube_km.unmasked_data[:,:,:])
+	cube_gauss_noise = np.array(cube_km.unmasked_data[:,:,:])
+	shape = np.shape(cube_gauss)
+
+	# Set up a cube for storing fitted parameters
+	param_cube = cube_gauss[0]
```

Comment:
Would this be better as a direct call to `np.zeros()` like `np.zeros(6, shape[1], shape[2])`?

Reply:
Definitely. Apparently I thought copying and pasting a parameter name six times was the way to go when I wrote this...